raiden_contracts/tests/test_unit_internals.py

```diff
@@ -0,0 +1,246 @@
+        create_channel_and_deposit,
+        get_accounts
+):
+
+    (A, B) = get_accounts(2)
+    deposit_A = 5
+    deposit_B = 7
+    withdraw_A = 3
+    channel_identifier = create_channel_and_deposit(A, B, deposit_A, deposit_B)
+    (signature_A, signature_B) = create_withdraw_signatures(
+        [A, B],
+        channel_identifier,
+        A,
+        withdraw_A,
+        token_network_test.address
+    )
+    token_network_test.functions.verifyWithdrawSignaturesPublic(
+        channel_identifier,
+        A,
+        B,
+        3,
+        signature_A,
+        signature_B
+    ).call()
+
+    with pytest.raises(TransactionFailed):
+        token_network_test.functions.verifyWithdrawSignaturesPublic(
+            channel_identifier,
+            A,
+            B,
+            3,
+            signature_B,
+            signature_A
+        ).call()
+    with pytest.raises(TransactionFailed):
+        token_network_test.functions.verifyWithdrawSignaturesPublic(
+            channel_identifier,
+            A,
+            B,
+            3,
+            signature_A,
+            fake_bytes(64)
+        ).call()
+    with pytest.raises(TransactionFailed):
+        token_network_test.functions.verifyWithdrawSignaturesPublic(
+            channel_identifier,
+            A,
+            B,
+            3,
+            fake_bytes(64),
+            signature_B
+        ).call()
+
+    recovered_address_A = token_network_test.functions.recoverAddressFromWithdrawMessagePublic(
+        channel_identifier,
+        A,
+        withdraw_A,
+        signature_A
+    ).call()
+    assert recovered_address_A == A
+
+    recovered_address_B = token_network_test.functions.recoverAddressFromWithdrawMessagePublic(
+        channel_identifier,
+        A,
+        withdraw_A,
+        signature_B
+    ).call()
+    assert recovered_address_B == B
+
+    with pytest.raises(TransactionFailed):
+        token_network_test.functions.recoverAddressFromWithdrawMessagePublic(
+            channel_identifier,
+            A,
+            withdraw_A,
+            fake_hex(64)
+        ).call()
+
+    assert recovered_address_A != \
+        token_network_test.functions.recoverAddressFromWithdrawMessagePublic(
+            channel_identifier,
+            B,
+            withdraw_A,
+            signature_A
+        ).call()
+
+    assert recovered_address_A != \
+        token_network_test.functions.recoverAddressFromWithdrawMessagePublic(
+            channel_identifier,
+            A,
+            1,
+            signature_A
+        ).call()
+
+    assert recovered_address_A != \
+        token_network_test.functions.recoverAddressFromWithdrawMessagePublic(
+            channel_identifier,
+            A,
+            withdraw_A,
+            signature_B
+        ).call()
+
+
+def test_recoverAddressFromBalanceProof(
+        token_network_test,
+        create_balance_proof_internal,
+        create_balance_proof_update_signature_internal,
```

Comment:
Why use `create_balance_proof_internal` and `create_balance_proof_update_signature_internal`  and not the already existent `create_balance_proof`, `create_balance_proof_update_signature` ?